### PR TITLE
Contact Form: Update default labels

### LINKED ISSUE #27429: Form Block - Bug: Multi-line text block without label in editor, shows "message" as label in published view.
If a user chooses not to fill the label, we shouldn't show a label in the published view.

Image

### PR DESCRIPTION
**Pull request by jcheringer**
Fixes #27429

#### Changes proposed in this Pull Request:
* Updates the default labels logic to allow fields without any label

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Create a post and a Form block
* Choose a variation and add other fields to the Form
* Check if the fields allow "empty" labels

### CODE REVIEW

**Review by digitalwaveride — DISMISSED** [review]
Nice João!
Things worked as advertised! The label did not show for me when I kept it empty.

The only thing I noticed is on the "Message sent" view, that a ":" was sneaking in, and I assume also in the email that would have ben sent. Do we need a character there? If that's the case, probably a "-" would make more sense.

image

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php [hunk 1]
```diff
@@ -3041,9 +3041,6 @@ public static function get_default_label_from_type( $type ) {
 			case 'textarea':
 				$str = __( 'Message', 'jetpack' );
 				break;
-			case 'checkbox':
```
Comment (CGastrell) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:3044]: Sorry I’m not familiar with the change, but why this removal?

Comment (jcheringer) [reply projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:3044]: In the editor, the checkbox field doesn't have a default label value. 
We need to remove this piece of code to match the same behavior in the backend

**Comment by jcheringer** [comment]
Thanks for testing it @digitalwaveride!
It should be fixed now.

**Review by CGastrell — APPROVED** [review]
Works as described!

One thing worth mentioning though: when using empty labels, public view will render empty tags, hence "skipping"/collapsing the line. This makes for (maybe) undesired less spacing between inputs. Not a blocker, just a mention.

Screenshot 2022-12-14 at 16 59 33

**Review by jeherve — APPROVED** [review]

**Pull request merged**